From 6ecdbb1db2a09f11f08958739751b63d25ceb596 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Fri, 9 Aug 2024 23:09:18 +0800
Subject: [PATCH 27/27] Reland^2 "linux/build: Enable fission for (symbol_level
 = 2) release builds as well"

This is a reland of commit b7da3afa43ecb6a1a4839c4f91e798a6d72b221f

The CL got reverted for breaking android-official, and a comment
on the original review mentioned that it also broke a few cros bots
like arm-generic-cq.

For the reland, this now also allows non-fission static symbol_level 2
for android and chromeos.

Original change's description:
> Reland "linux/build: Enable fission for (symbol_level = 2) release builds as well"
>
> This is a reland of commit c031e3aa3a7a0ad71715655e140ba52ffa82a08d
> The reland removes an assert on macOS (where is_debug_fission isn't
> used -- darwin toolchains have always used something like fission,
> but under a different name).
>
> Original change's description:
> > linux/build: Enable fission for (symbol_level = 2) release builds as well
> >
> > Looks like full-symbol non-component builds are getting too big for
> > debug info even in release builds.
> >
> > Turn on fission by default on linux in release builds as well.
> >
> > This might affect official builds, where dump_syms etc might not
> > handle fission yet. If this is a problem, we'll have to fix this
> > at some point, but short-term we can not do this for official
> > builds in that case. (zequanwu@ says CrOS enabled fission already,
> > encountered dump_syms issues, and those already got fixed.
> > And a linux-official try run is green. We'll see!)
> >
> > (Component builds don't need fission yet, but for simplicity I'm
> > enabling fission independent of is_component_build.)
> >
> > Bug: 353915727
> > Cq-Include-Trybots: luci.chromium.try:linux-official
> > Change-Id: I0df4031726f024e60646e43285d85b150a778d24
> > Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5730815
> > Auto-Submit: Nico Weber <thakis@chromium.org>
> > Reviewed-by: Zequan Wu <zequanwu@google.com>
> > Commit-Queue: Zequan Wu <zequanwu@google.com>
> > Cr-Commit-Position: refs/heads/main@{#1331406}
>
> Bug: 353915727
> Change-Id: Idf9df6d94ab3a8c7b4e3f6d8bbb7d50fd2066b13
> Cq-Include-Trybots: luci.chromium.try:linux-official
> Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5731601
> Auto-Submit: Nico Weber <thakis@chromium.org>
> Commit-Queue: Nico Weber <thakis@chromium.org>
> Commit-Queue: Kenichi Ishibashi <bashi@chromium.org>
> Reviewed-by: Kenichi Ishibashi <bashi@chromium.org>
> Cr-Commit-Position: refs/heads/main@{#1331479}

Bug: 353915727
Change-Id: I71bbfff81f9d2e04d3f457e25b90d5da234b25fb
Cq-Include-Trybots: luci.chromium.try:linux-official
Cq-Include-Trybots: luci.chromium.try:android-official
Cq-Include-Trybots: luci.chromium.try:chromeos-amd64-generic-rel
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5753311
Commit-Queue: Zequan Wu <zequanwu@google.com>
Reviewed-by: Zequan Wu <zequanwu@google.com>
Auto-Submit: Nico Weber <thakis@chromium.org>
Commit-Queue: Nico Weber <thakis@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1335843}
Co-authored-by: Nico Weber <thakis@chromium.org>
---
 build/config/compiler/compiler.gni | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/build/config/compiler/compiler.gni b/build/config/compiler/compiler.gni
index e9acde43a5..36645382cf 100644
--- a/build/config/compiler/compiler.gni
+++ b/build/config/compiler/compiler.gni
@@ -73,8 +73,8 @@ declare_args() {
   # with some utilities such as icecc and ccache. Requires gold and
   # gcc >= 4.8 or clang.
   # http://gcc.gnu.org/wiki/DebugFission
-  use_debug_fission = is_debug && !is_android && !is_fuchsia && !is_apple &&
-                      !is_win && use_lld && cc_wrapper == ""
+  use_debug_fission = !is_android && !is_fuchsia && !is_apple && !is_win &&
+                      use_lld && cc_wrapper == ""
 
   # Enables support for ThinLTO, which links 3x-10x faster than full LTO. See
   # also http://blog.llvm.org/2016/06/thinlto-scalable-and-incremental-lto.html
@@ -312,9 +312,9 @@ use_debug_fission = use_debug_fission && symbol_level == 2
 if (forbid_non_component_debug_builds) {
   assert(
       symbol_level != 2 || current_toolchain != default_toolchain ||
-          is_component_build || !is_debug || is_ios || use_debug_fission ||
-          host_os == "win",
-      "Can't do non-component debug builds at symbol_level=2 without use_debug_fission=true")
+          is_component_build || is_fuchsia || is_apple || use_debug_fission ||
+          is_android || is_chromeos || host_os == "win",
+      "Can't do non-component builds at symbol_level=2 without use_debug_fission=true")
 }
 
 # TODO(crbug.com/40230692) For Windows, to assemble lzma_sdk's assembly files,
-- 
2.46.0

